vpp/oclpostprocess_transform.cpp

```diff
@@ -0,0 +1,281 @@
+/*
+ *  oclpostprocess_transform.cpp - opencl based flip and rotate
+ *
+ *  Copyright (C) 2016 Intel Corporation
+ *    Author: Jia Meng<jia.meng@intel.com>
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public License
+ *  as published by the Free Software Foundation; either version 2.1
+ *  of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free
+ *  Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ *  Boston, MA 02110-1301 USA
+ */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "oclpostprocess_transform.h"
+#include "vaapipostprocess_factory.h"
+#include "common/common_def.h"
+#include "common/log.h"
+#include "ocl/oclcontext.h"
+#include "vaapi/vaapiutils.h"
+#include "vpp/oclvppimage.h"
+
+namespace YamiMediaCodec {
+
+YamiStatus
+OclPostProcessTransform::process(const SharedPtr<VideoFrame>& src,
+    const SharedPtr<VideoFrame>& dst)
+{
+    YamiStatus status = ensureContext("transform");
+    if (status != YAMI_SUCCESS)
+        return status;
+
+    if (src->fourcc != YAMI_FOURCC_NV12 || dst->fourcc != YAMI_FOURCC_NV12) {
+        ERROR("only support transform of NV12 video frame");
```

Comment:
could we add some width height check here? 320x240 can't rotate 320x240.. 320x240 can't flip to 120x120...